plugins/registry/functions.go

```diff
@@ -100,9 +109,21 @@ func pushToRegistry(appName string, tag int, imageID string, imageRepo string) e
 		return errors.New("Unable to tag image")
 	}
 
-	// For the future, we should also add the ability to create the remote repository
-	// This is only really important for registries that do not support creation on push
-	// Examples include AWS and Quay.io
+	extraTags := getRegistryPushExtraTagsForApp(appName)
+	if extraTags != "" {
+		extraTagsArray := strings.Split(extraTags, ",")
+		for _, extraTag := range extraTagsArray {
+			extraTagImage := fmt.Sprintf("%s%s:%s", registryServer, imageRepo, extraTag)
+			common.LogVerboseQuiet(fmt.Sprintf("Tagging %s as %s in registry format", imageRepo, extraTag))
+			if !dockerTag(imageID, extraTagImage) {
+				return errors.New(fmt.Sprintf("Unable to tag image as %s", extraTag))
+			}
```

Comment:
Add a defer after this line removing the extra tags.

Reply:
yeah, I'll get back to it end of this week, been busy doing other stuff at work :)

Reply:
Sounds good!

Reply:
Hi @josegonzalez, I've added the `dockerRemoveTag` function that does the trick.

Reply:
It basically works, the extra tag added is removed/untagged:

```
-----> Pushing image to registry
       Bumping tag to 4
       Retrieving image info for app
       Tagging ktaraszk/node-js-getting-started:4 in registry format
       Tagging ktaraszk/node-js-getting-started as latest in registry format
       Pushing harbor.local/ktaraszk/node-js-getting-started:latest
The push refers to repository [harbor.local/ktaraszk/node-js-getting-started]
e9f706ddc04b: Preparing
cff76183993c: Preparing
c3c69f05393e: Preparing
8d13edf8ce8f: Preparing
ee985203e4ac: Preparing
27d43a296abf: Preparing
1266b6e15d12: Preparing
c3c81a53fa22: Preparing
b8d53d184a96: Preparing
423d4343f554: Preparing
824ea57cb41b: Preparing
86395255d96f: Preparing
82808dc54797: Preparing
e2ded895d452: Preparing
8e87ff28f1b5: Preparing
27d43a296abf: Waiting
1266b6e15d12: Waiting
c3c81a53fa22: Waiting
b8d53d184a96: Waiting
423d4343f554: Waiting
824ea57cb41b: Waiting
86395255d96f: Waiting
82808dc54797: Waiting
e2ded895d452: Waiting
8e87ff28f1b5: Waiting
e9f706ddc04b: Pushed
8d13edf8ce8f: Pushed
ee985203e4ac: Pushed
c3c69f05393e: Pushed
27d43a296abf: Layer already exists
1266b6e15d12: Layer already exists
b8d53d184a96: Layer already exists
c3c81a53fa22: Layer already exists
423d4343f554: Layer already exists
824ea57cb41b: Layer already exists
82808dc54797: Layer already exists
86395255d96f: Layer already exists
e2ded895d452: Layer already exists
8e87ff28f1b5: Layer already exists
cff76183993c: Pushed
latest: digest: sha256:a6b53c2b7c79ba770013edaf02335ca31ac9c11efb83a71da9517125439ea14f size: 3454
       Untagging extra tag harbor.local/ktaraszk/node-js-getting-started:latest
       Pushing harbor.local/ktaraszk/node-js-getting-started:4
The push refers to repository [harbor.local/ktaraszk/node-js-getting-started]
e9f706ddc04b: Preparing
cff76183993c: Preparing
c3c69f05393e: Preparing
8d13edf8ce8f: Preparing
ee985203e4ac: Preparing
27d43a296abf: Preparing
1266b6e15d12: Preparing
c3c81a53fa22: Preparing
b8d53d184a96: Preparing
423d4343f554: Preparing
824ea57cb41b: Preparing
86395255d96f: Preparing
82808dc54797: Preparing
e2ded895d452: Preparing
8e87ff28f1b5: Preparing
27d43a296abf: Waiting
1266b6e15d12: Waiting
c3c81a53fa22: Waiting
b8d53d184a96: Waiting
423d4343f554: Waiting
824ea57cb41b: Waiting
86395255d96f: Waiting
82808dc54797: Waiting
e2ded895d452: Waiting
8e87ff28f1b5: Waiting
ee985203e4ac: Layer already exists
e9f706ddc04b: Layer already exists
8d13edf8ce8f: Layer already exists
cff76183993c: Layer already exists
c3c69f05393e: Layer already exists
c3c81a53fa22: Layer already exists
27d43a296abf: Layer already exists
1266b6e15d12: Layer already exists
b8d53d184a96: Layer already exists
423d4343f554: Layer already exists
86395255d96f: Layer already exists
82808dc54797: Layer already exists
824ea57cb41b: Layer already exists
8e87ff28f1b5: Layer already exists
e2ded895d452: Layer already exists
4: digest: sha256:a6b53c2b7c79ba770013edaf02335ca31ac9c11efb83a71da9517125439ea14f size: 3454
       Image harbor.local/ktaraszk/node-js-getting-started:4 pushed
-----> Deploying node-js-getting-started via the docker-local scheduler...
```